python/test/unit/language/test_core_amd.py

```diff
@@ -1492,6 +1492,8 @@ def kernel(X, stride_xm, stride_xn,
                           for allow_tf32 in [True, False]
                           for in_dtype, out_dtype in [('float16', 'float16'),
                                                       ('bfloat16', 'float32'),
+                                                      ('float8e5m2fnuz', 'float32'),
+                                                      ('float8e4m3fnuz', 'float32'),
```

Comment:
@scxiao 
I've added following tests in `test_dot` testsuite.

These are no mixed precision cases, just `fp8xfp8 - > fp32` and `bf8xbf8 - > fp32`, I assume I had to add these mixed precision cases in `test_gemm_fp816_mixed_inputs` tesuite.
But I can not do it at the moment, because there are no needed data types yet triton :(

Because of this problem with element types, current fp8 tests in `test_dot` testsuite are kind of broken on mi300, but they should work on older GPU, because everything is casted to supported types, we need to change these data types later, after you merge your fp8 support.


Idea behind tests I've added is following:
If we need to test fp8 dot, we pass `fp32` input tensors and "effective" data type of these tensors inside kernel.
If effective data type is fp8, kernel casts tensor to given fp8 data type after loading it from global memory.

With this approach we do not need fp8 support from torch and numpy to test triton fp8 implementation.

Reply:
Hi @binarman, I ran this PR on a MI300, it seems like these two types are failed for the unit tests, what should I change to make it pass?

Reply:
never mind, I put some fix and got all tests passed.